src/compute/src/extensions/arrange.rs

```diff
@@ -255,13 +262,9 @@ where
                 old_allocations = allocations;
             }
         });
-    Arranged {
-        trace: arranged.trace,
-        stream,
-    }
```

Comment:
Your understanding is correct. The problem is that we need to attach the logging operator to the stream potentially after someone else attached themselves to the stream already, so we need to clone anyways. The clones are cheap, because we send `Rc<Batch>`, so I'm not worried about this.

We can't turn the operator into a `sink` because `sink` doesn't give us access to the `OperatorInfo`, which we need to determine the operator's address to be able to activate it from the outside. We could rewrite it in terms of `builder_rc`.

Reply:
I think `unary` is fine. Though should we maybe remove the `output.session(&time).give_container(&mut buffer);` to not make readers think that the output will be used?